rbi/core/enum.rbi

```diff
@@ -7,6 +7,13 @@ class T::Enum
   sig {returns(T::Array[T.experimental_attached_class])}
   def self.values; end
 
+  # Convert from serialized value to enum instance.
+  #
+  # Note: It would have been nice to make this method final before people started overriding it.
+  # Note: Failed CriticalMethodsNoRuntimeTypingTest
```

Comment:
Smaller nit: these comments are just copy/pasted from `from_serialized` but these probably aren't relevant here.

Larger nit, inspired by the comment itself: should we make `try_deserialize` a final method now since we've just introduced it?

Reply:
I thought about it, but I'm not sure how final works across libraries, because to get it fully working you have to run some `T::Configuration.enable_*` shenanigans and I'm not sure that I want to be the first person to try to figure out the consequences of that. Also `deserialize` is by far the more important method, because it's the one that `T::Props` calls, so we get diminishing utility from making this unused helper final.